src/muse/mca.py

```diff
@@ -538,94 +538,44 @@ def find_equilibrium(
 
     from numpy import ones
 
-    from muse.utilities import future_propagation
-
     market = market.copy(deep=True)
     if excluded_commodities:
         included = ~market.commodity.isin(excluded_commodities)
     else:
         included = ones(len(market.commodity), dtype=bool)
 
-    market["updated_prices"] = drop_timeslice(market.prices.copy())
     prior_market = market.copy(deep=True)
     converged = False
-    equilibrium_sectors = sectors
-    for i in range(maxiter):
-        market["prices"] = drop_timeslice(market.updated_prices)
+    iteration = 0
+    while iteration < maxiter and not converged:
         prior_market, market = market, prior_market
         market.consumption[:] = 0.0
         market.supply[:] = 0.0
         market, equilibrium_sectors = single_year_iteration(market, sectors)
 
-        check_demand_fulfillment(market.sel(commodity=included), tol_unmet_demand)
+        if maxiter == 1 or not equilibrium:
+            converged = True
+            break
 
-        equilibrium_reached = check_equilibrium(
+        # Check convergence criteria
+        check_demand_fulfillment(market.sel(commodity=included), tol_unmet_demand)
```

Comment:
This function returns `True` or `False`, but it is not assigned or used for anything, so it seems to me there's something missing here. As this was the behaviour before, let's leave it as it is, but seems odd to use this just to log the problematic behaviour.

Reply:
Yes I agree. See #410. It's also not very noticeable when the warning does get logged as it gets hidden among the barrage of other log messages. I'll leave it as it is for now, but might be something to revisit in the future.